server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -172,10 +180,14 @@ public int getNumOpenConnections()
         private final AtomicLong byteCount = new AtomicLong(0);
         private final BlockingQueue<InputStream> queue = new LinkedBlockingQueue<>();
         private final AtomicBoolean done = new AtomicBoolean(false);
+        private final AtomicReference<String> fail = new AtomicReference<>();
 
         @Override
         public ClientResponse<InputStream> handleResponse(HttpResponse response)
         {
+          checkQueryTimeout();
```

Comment:
they contain specific error message which would need to be repeated at all call sites. also, they are used in a relatively small scope and look ok.